File: CS/1720_Decode_XORed_Array.cs
public class Solution {
    public int[] Decode(int[] encoded, int first) {
        int[] result = new int[encoded.Length+1];
        result[0] = first;
        for(int i = 1; i < result.Length; i++){
            result[i] = result[i-1]^encoded[i-1];
        }
        return result;
    }
}

File: Cpp/1720_Decode_XORed_Array.cpp
class Solution {
public:
    vector<int> decode(vector<int>& encoded, int first) {
        vector<int> result;
        int temp = first;
        result.push_back(temp);
        for(int i = 0; i < encoded.size(); i++){
            temp ^= encoded[i];
            result.push_back(temp);
        }
        return result;
    }
};
